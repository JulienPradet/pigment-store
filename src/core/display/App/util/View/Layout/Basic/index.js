import React from 'react'
import styles from './layout.m.css'

const Layout = ({nav, horizontalNav, children}) => <div className={styles.container}>
  <div className={styles.nav}>{nav}</div>
  <div className={styles.contentContainer}>
    <div className={styles.horizontalNav}>{horizontalNav}</div>
    <div className={styles.content}>{children}</div>
  </div>
</div>

export default Layout
