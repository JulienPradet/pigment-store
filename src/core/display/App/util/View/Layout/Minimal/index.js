import React from 'react'
import styles from './layout.m.css'

const Layout = class Layout extends React.Component {
  constructor () {
    super()
    this.state = {
      showHiddenNav: false
    }
    this.toggleHiddenNav = this.toggleHiddenNav.bind(this)
  }

  toggleHiddenNav () {
    this.setState((state) => ({
      showHiddenNav: !state.showHiddenNav
    }))
  }

  render () {
    const {nav, hiddenNav, returnNav, children} = this.props
    return (
      <div className={styles.layoutContainer}>
        <div className={styles.navWrapper}>
          <div className={styles.navContainer}>
            <div className={styles.backButton}>{returnNav}</div>
            <div className={styles.mainNav}>{nav}</div>
            <div className={styles.hiddenNavButton}>
              <button onClick={this.toggleHiddenNav}>
                <i className='material-icons'>settings</i>
              </button>
            </div>
          </div>
          {this.state.showHiddenNav && hiddenNav}
        </div>
        <div className={styles.contentContainer}>
          {children}
        </div>
      </div>
    )
  }
}

export default Layout
