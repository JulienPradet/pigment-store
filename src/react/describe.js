import describe from './definition/Component'

export default describe
