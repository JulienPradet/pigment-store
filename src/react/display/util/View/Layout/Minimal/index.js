import React from 'react'
import styles from './layout.css'

const Layout = ({nav, hiddenNav, returnNav, children}) => <div>
  <div className={styles.navContainer}>
    <div className={styles.backButton}>{returnNav}</div>
    <div>{nav}</div>
    <div>{hiddenNav}</div>
  </div>
  <div>{children}</div>
</div>

export default Layout
