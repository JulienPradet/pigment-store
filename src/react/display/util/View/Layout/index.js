import React from 'react'
import styles from './layout.css'

export default ({nav, horizontalNav, children}) => <div className={styles.container}>
  <div className={styles.nav}>{nav}</div>
  <div className={styles.contentContainer}>
    <div className={styles.horizontalNav}>{horizontalNav}</div>
    <div className={styles.content}>{children}</div>
  </div>
</div>
