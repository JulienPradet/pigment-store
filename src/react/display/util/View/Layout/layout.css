body {
  margin: 0;
  background: #ece7f0;
  border-right: 0.75em solid #a6cf39;
  border-bottom: 0.75em solid #a6cf39;
  box-sizing: border-box;
  height: 100vh;
  width: 100%;
  overflow: auto;
  font-family: 'PT Sans', sans-serif;
  line-height: 1.5em;
  font-size: 14pt;
  color: #39354b;
}

pre {
  font-size: 11pt;
}

a {
  color: #a6cf39;
}

button, input {
  font-size: inherit;
}

p:first-child {
  margin-top: 0;
}
p:last-child {
  margin-bottom: 0;
}

:global .material-icons {
  font-size: inherit !important;
  vertical-align: middle;
}

.container {
  display: flex;
  flex-direction: row;
  height: calc(100vh - 0.75em);
  overflow: hidden;
}

.nav {
  flex: 1;
  background: #fff;
  overflow: auto;
}

.contentContainer {
  flex: 4;
  overflow: auto;
}

.content {
  margin: 2em 2em;
}
