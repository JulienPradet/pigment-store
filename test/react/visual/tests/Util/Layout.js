import React from 'react'
import describe from 'pigment-store/react/describe'
import Layout from '../../../../../src/core/display/App/util/View/Layout/Basic/index'

export default describe('Layout', Layout)
  .featureJsx(
    'Default',
    () => (
      <Layout nav='Left navigation' horizontalNav='Top navigation'>
        Content
      </Layout>
    )
  )
