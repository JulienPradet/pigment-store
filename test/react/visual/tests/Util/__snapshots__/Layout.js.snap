exports[`Layout Default 1`] = `
<div
  className="container">
  <div
    className="nav">
    Left navigation
  </div>
  <div
    className="contentContainer">
    <div
      className="horizontalNav">
      Top navigation
    </div>
    <div
      className="content">
      Content
    </div>
  </div>
</div>
`;
